src/l500/l500-depth.cpp

```diff
@@ -163,6 +163,60 @@ namespace librealsense
         return std::make_shared<timestamp_composite_matcher>(matchers);
     }
 
+
+    // If the user did not ask for IR, The open function will add it anyway. 
+    // This class filters the IR frames is this case so they will not get to the user callback function
+    class frame_filter : public rs2_frame_callback
```

Comment:
Please add a comment that this is a workaround and that, theoretically, the processing blocks should be taking care of all this without the need to manually add the IR stream

Reply:
Added